src/profile.c

```diff
@@ -532,11 +530,61 @@ static inline void printOutput(int fileid, int line, BOOL exec, BOOL visited)
     }
 }
 
+// Mark line as visited, and return true if the line has been previously
+// visited (executed)
+BOOL markVisited(int fileid, UInt line)
+{
+    // Some STATs end up without a file or line -- do not output these
+    // as they would just confuse the profile generation later.
+    if (fileid == 0 || line == 0) {
+        return TRUE;
+    }
+
+    if (LEN_PLIST(profileState.visitedStatements) < fileid ||
+        !ELM_PLIST(profileState.visitedStatements, fileid)) {
+        AssPlist(profileState.visitedStatements, fileid,
+                 NEW_PLIST(T_PLIST, 0));
+    }
+
+    Obj linelist = ELM_PLIST(profileState.visitedStatements, fileid);
+
+    if (LEN_PLIST(linelist) < line || !ELM_PLIST(linelist, line)) {
+        AssPlist(linelist, line, True);
+        return FALSE;
+    }
+    return TRUE;
+}
+
+// Return TRUE is Stat has been visited (executed) before
+BOOL visitedStat(Stat stat)
```

Comment:
This function is never called, maybe it can be removed? Or is it supposed to be called from somewhere after all?